docs/source/explanation/templates.rst

```diff
@@ -3,29 +3,29 @@
 What is a template?
 ------------------------
 
-A YAML template (see :ref:`explanation_yaml`) in HTTomo is an interface to a method which can be used to build a pipeline of tasks or a process list that will be executed. See more on :ref:`explanation_process_list`
-The template provides communication with a method by setting input/output entries and also additional parameters, if required.
+A YAML template (see :ref:`explanation_yaml`) is a textual interface to a method, which can be executed in HTTomo. 
+The template provides a communication with a chosen method by setting its input/output entries and also additional parameters, if required.
 
-Let's consider this simple template for a median filter from the `TomoPy <https://tomopy.readthedocs.io/en/stable/api/tomopy.misc.corr.html#tomopy.misc.corr.median_filter3d>`_ package.
+The combination of YAML templates results in a processing list, also called a pipeline. See more on :ref:`explanation_process_list`
+
+As a simple template example, let's consider the template for the median filter from the `TomoPy <https://tomopy.readthedocs.io/en/stable/api/tomopy.misc.corr.html#tomopy.misc.corr.median_filter3d>`_ package.
 
 .. code-block:: yaml
     
-    tomopy.misc.corr:
-      median_filter3d:
-        data_in: tomo
-        data_out: tomo
+    - method: median_filter3d
+      module_path: tomopy.misc.corr
+      parameters:
         size: 3
 
-The first line :code:`tomopy.misc.corr` specifies the module in TomoPy library and the second :code:`median_filter3d` is the 
-name of the method in that module. HTTomo will interpret this in Python
-by importing the method:
+The first two lines are self-explanatory: the method's name and path to the module. HTTomo will interpret this in Python
+by importing the method as:
 
 .. code-block:: python
 
     from tomopy.misc.corr import median_filter3d
 
-After the method's name follows the list of parameters for that method. Parameters :code:`data_in` 
-and :code:`data_out` define the names of the input and the output datasets respectively. Then :code:`size` 
-is specific to the method parameter.
+The list of parameters for that method is given in the *parameters* list.
```

Comment:
This isn't really a big deal, it's kinda trying to guess what users would think regarding wording, so feel free to take it or leave it.

The value of the `parameters` field is more a map of key-value pairs than a list (it becomes a python `dict`), and I'm not sure if referring to it as a "list" could confuse people.

I tend to use the term "field" when referring to a key in a mapping in YAML since it doesn't imply a particular python data structure in the same way "list" does, so maybe something like the following could be less ambiguous:

```suggestion
The set of parameter values for that method is given in the *parameters* field.
```